lib/Models/RegionParameter.js

```diff
@@ -96,7 +96,11 @@ RegionParameter.prototype.findRegionByID = function(regionID) {
     }
 
     return regionProvider.regions.filter(function(region) {
-        return region.id === regionID;
+        if (typeof region.id === 'string' && typeof regionID === 'string') {
```

Comment:
Argh! This used to fix the problem described, but you're right, that now works without my change. I've removed my change. Note that it's still comparing Australia to australia and returning no match... but the country is still selected, so I guess there's something that has changed elsewhere.